src/share/configuration_core.hpp

```diff
@@ -166,6 +170,39 @@ class configuration_core final {
     return v;
   }
 
+  std::vector<std::pair<krbn::key_code, std::vector<krbn::key_code>>> get_key_code_patterns_from_json_object(const nlohmann::json& json) {
```

Comment:
I'll need to check for conflicts between `simple_modifications` and `compound_modifications`